src/java/org/apache/cassandra/net/Message.java

```diff
@@ -1287,7 +986,7 @@ private Map<ParamType, Object> deserializeParams(DataInputPlus in, int version)
          */
         private Map<ParamType, Object> extractParams(ByteBuffer buf, int readerIndex, int version) throws IOException
         {
-            long count = version >= VERSION_40 ? getUnsignedVInt(buf, readerIndex) : buf.getInt(readerIndex);
+            long count = getUnsignedVInt(buf, readerIndex);
```

Comment:
Nit: The JavaDoc for this method has a mention of `post-4.0` that we can remove. Also, the class JavaDoc for `MessageSerializer` has a list of "Legacy notes" that can also be removed.

Reply:
done